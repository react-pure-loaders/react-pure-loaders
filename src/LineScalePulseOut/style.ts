import { keyframes } from '@emotion/react';
import { globalAnimation, lines } from '../mixins';
import delay from '../utilities';

const lineScalePulseOut = keyframes({
  '0%': {
    transform: 'scaley(1.0)',
  },
  '100%': {
    transform: 'scaley(1.0)',
  },
  '50%': {
    transform: 'scaley(0.4)',
  },
});

export default (color?: string) => ({
  ' > div': {
    ...lines(color),
    ...globalAnimation(),
    animation: `${lineScalePulseOut} 0.9s ${delay('0.2s', 3, 0)} infinite cubic-bezier(.85,.25,.37,.85)`,
    display: 'inline-block',

    '&:nth-of-type(2), &:nth-of-type(4)': {
      animationDelay: `${delay('0.2s', 3, 1)} !important`,
    },

    '&:nth-of-type(1), &:nth-of-type(5)': {
      animationDelay: `${delay('0.2s', 3, 2)} !important`,
    },
  },
});
