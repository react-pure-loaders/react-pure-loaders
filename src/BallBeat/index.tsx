/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
