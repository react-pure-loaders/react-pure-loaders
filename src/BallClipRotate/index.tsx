/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
