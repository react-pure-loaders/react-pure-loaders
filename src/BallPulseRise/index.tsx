/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
