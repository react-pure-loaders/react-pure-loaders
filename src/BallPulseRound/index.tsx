/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
