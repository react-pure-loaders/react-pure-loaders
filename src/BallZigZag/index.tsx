/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
