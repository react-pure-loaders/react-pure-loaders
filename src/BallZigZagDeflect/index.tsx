/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
