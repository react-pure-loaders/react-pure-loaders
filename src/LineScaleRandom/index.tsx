/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
