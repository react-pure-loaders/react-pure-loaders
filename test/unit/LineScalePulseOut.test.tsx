import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import LineScalePulseOut from '../../src/LineScalePulseOut';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<LineScalePulseOut>', () => {
    afterEach(cleanup);

    test('LineScalePulseOut should match snapshot', () => {
        const { container } = render(<LineScalePulseOut loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('LineScalePulseOut should have default color', () => {
        const { container } = render(<LineScalePulseOut loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', PRIMARY_COLOR, { target: '> div' });
    });

    test('LineScalePulseOut should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<LineScalePulseOut color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', color, { target: '> div' });
    });

    test('LineScalePulseOut should have no children', () => {
        const { container } = render(<LineScalePulseOut loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('LineScalePulseOut should have three children', () => {
        const { container } = render(<LineScalePulseOut loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(6);
    });
});
