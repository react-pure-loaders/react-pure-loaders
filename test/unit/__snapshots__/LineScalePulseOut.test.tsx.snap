// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineScalePulseOut> LineScalePulseOut should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  100% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  50% {
    -webkit-transform: scaley(0.4);
    -moz-transform: scaley(0.4);
    -ms-transform: scaley(0.4);
    transform: scaley(0.4);
  }
}

.emotion-0>div {
  border-radius: 2px;
  height: 35px;
  margin: 2px;
  width: 4px;
  background-color: #ffffff;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation: animation-0 0.9s -0.60s infinite cubic-bezier(.85,.25,.37,.85);
  animation: animation-0 0.9s -0.60s infinite cubic-bezier(.85,.25,.37,.85);
  display: inline-block;
}

.emotion-0>div:nth-of-type(2),
.emotion-0>div:nth-of-type(4) {
  -webkit-animation-delay: -0.40s!important;
  animation-delay: -0.40s!important;
}

.emotion-0>div:nth-of-type(1),
.emotion-0>div:nth-of-type(5) {
  -webkit-animation-delay: -0.20s!important;
  animation-delay: -0.20s!important;
}

<div
  class="emotion-0"
>
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;
